### DISCUSSION TITLE
Is there a way to use Function Keys inside Blink Code?

### DISCUSSION THREAD

**Discussion by aauren** [blinksh/blink#2031] [Support]
I have function keys currently mapped to the Command key + Number on my iPad and use the Magic Keyboard with my iPad. But this doesn’t appear to work when I’m inside a VS Code window in the blink shell. My guess is that Blink is preferring to send the Command key to Code natively, which makes sense for a lot of keys. However function keys, like F2 are super helpful to have when using code.

Is there a way to get Blink Shell to produce function keys when in Code?

Additionally, is there a way to use the Delete key inside code? I tried doing Command+Delete, but that didn’t appear to work either.

**Comment by rothgar** [comment]
Hi Aaron 👋 I was just looking for the same answer and of course I found you asked this question a month before me. 😅

**Comment by aauren** [comment]
Any thoughts here @carloscabanero / @yury ?

**Reply by carloscabanero** [reply]
Sorry I missed this one. They won't work at the moment. I thought that maybe doing a custom Key Press maybe would, but that was not the case. 

The issue is that it collides with other Shell cases as mentioned before. To really make Blink Code reach its full potential, we need to separate it from the Shell itself. When we launched Code, we didn't know if people would be interested in a local editor. So we adapted some cool scenarios but still left some rough corners.

We can get there, but we need more resources - people purchasing Blink Plus.

**Reply by aauren** [reply]
Thanks for the response @carloscabanero! And thanks for making Blink!

FWIW I'm a subscriber. However, I do have to say that I've been using it less and less because of some of the rough edges. I'm still very excited that something like Blink exists out there, but I think one of the big advantages of VSCode is keeping your interface the same everywhere you go. Having to keep in my mind things like "I can't rename things by pressing "F2"" constantly knocks me out of my rhythm.

I'm worried that you'll lose people if you aren't able to get enough support to round out these rough corners over time.

Let me know if there are other ways to help! Maybe doing feature bounties or something like that?

**Reply by aauren** [reply]
Hey @carloscabanero just wanted to check in to see if there are any updates here? Any chance that there is a road to feature sponsorship on this feature? Or any chance that it has moved up on the roadmap on its own to be addressed in the near future?

**Comment by aauren** [comment]
Checking in again @carloscabanero as my Blink subscription is coming back up again. Is there any plan to make this better in the code integration?

**Reply by carloscabanero** [reply]
I have tried to fix this and other issues and except for really small things there isn't much we can do within the Shell app. On Blink 26 I want to start the path towards giving Code its own separate app. It won't be easy either because it took us a while to get the "browser" permissions, but I don't see other way forward. 

The path looks a bit like this:
- I will notify everyone on next version and start a signing page of some sort to see if there is real interest. Unless I can get enough users, I won't go through the whole process.
- We will start with a TestFlight version with those people who sign up.
- It would be awesome if we could get into some kind of partnership with platforms like GitPod, as that would help maintaining the app as well.
- Like Shell, the plan is that Code will be fully open source.
- First release would be for 27.

Hope that makes sense.

**Reply by aauren** [reply]
Thanks for the explanation!
